compiler/semtypes.nim

```diff
@@ -1042,6 +1043,35 @@ proc semBlockType(c: PContext, n: PNode, prev: PType): PType =
 proc semGenericParamInInvocation(c: PContext, n: PNode): PType =
   result = semTypeNode(c, n, nil)
 
+proc semRecordTypeAux(c: PContext, n: PNode, check: var IntSet) =
```

Comment:
That looks like you copied it from somewhere. ;-) You can make the original a template perhaps and instantiate it here. Also your code is wrong ``case`` in objects can introduce different AST kinds (which is why the internal structure is so weird to begin with...).